frontend/src/composables/useRequestFee.ts

```diff
@@ -32,11 +38,24 @@ export function useRequestFee(
       console.error(errorMessage);
       error.value = errorMessage ?? 'Unknown failure.';
     }
+
+    loading.value = false;
   };
 
-  watch([rpcUrl, requestManagerAddress, requestAmount], updateRequestFeeAmount, {
-    immediate: true,
-  });
+  const cb = computed(() =>
```

Comment:
There is nothing reactive inside, so you don't need a computed property here. I guess that happened because you had the debounced parameter as a ref before. I would just move the call down inside the watch. 

Also, I'd like to mention that I think it is better to have full names (callback instead of cb), because even if in most cases the meaning is clear it might not be clear to everyone. So it's just a small difference but provides better readability imo.

Reply:
True, true and true. Will do the necessary changes for this!